libs/cookies/src/lib/directives/has-cookie/has-cookie.directive.ts

```diff
@@ -0,0 +1,175 @@
+import {
+	ChangeDetectorRef,
+	ComponentRef,
+	Directive,
+	EmbeddedViewRef,
+	Inject,
+	Input,
+	Optional,
+	TemplateRef,
+	Type,
+	ViewContainerRef,
+} from '@angular/core';
+import { Subject, tap, takeUntil, combineLatest, map } from 'rxjs';
+import { flatten } from 'lodash';
+
+import { NgxHasCookieConfiguration } from '../../types';
+import { NgxCookieService } from '../../services';
+import { NgxCookiesFallBackComponent } from '../../abstracts';
+import { NgxCookiesFallbackComponentToken } from '../../tokens';
+
+/**
+ * A structural directive that provides a way to render UI elements based on whether a (set of) cookie(s) have been accepted by the user.
+ */
+@Directive({
+	selector: '[hasCookie]',
+	standalone: true,
+})
+export class NgxHasCookieDirective {
+	/**
+	 * The destroyed state of the directive
+	 */
+	private destroyed$: Subject<void>;
+
+	/**
+	 * The needed templateRefs
+	 */
+	private thenTemplateRef: TemplateRef<any> | null = null;
+	private thenViewRef: EmbeddedViewRef<any> | null = null;
+	private elseTemplateRef: TemplateRef<any> | null = null;
+	private elseViewRef: EmbeddedViewRef<any> | null = null;
+
+	/**
+	 * The ref of the component we wish to render as a fallback
+	 */
+	private componentRef: ComponentRef<NgxCookiesFallBackComponent>;
+
+	/**
+	 * The list of cookies we need to check
+	 */
+	private cookies: NgxHasCookieConfiguration[] = [];
+
+	/**
+	 * A cookie or list of cookies the item should have
+	 */
+	@Input() public set hasCookie(cookie: NgxHasCookieConfiguration | NgxHasCookieConfiguration[]) {
+		this.cookies = Array.isArray(cookie) ? cookie : [cookie];
+		this.updateView();
+	}
+
+	/**
+	 * The else template in case the cookie is not accepted
+	 */
+	@Input() public set hasCookieElse(ngTemplate: TemplateRef<any>) {
+		this.elseTemplateRef = ngTemplate;
+		this.elseViewRef = null;
+		this.updateView();
+	}
+
+	constructor(
+		templateRef: TemplateRef<any>,
```

Comment:
Yes, this allows you to prevent anything outside the constructor from using it. 

It's been explained to me once why the templateRef was done this way in the original *ngIf implementation, but don't ask me why because I've forgotten 🙈